cmake/mainSetup.cmake

```diff
@@ -0,0 +1,22 @@
+# In this file we configure some CMake settings we do not want to make visible directly in the main
+# CMakeLists.txt file.
+
+include(GNUInstallDirs)
+include(CheckFunctionExists)
+
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG ${CMAKE_BINARY_DIR}/bin)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE ${CMAKE_BINARY_DIR}/bin)
+
+
+if (UNIX)
+    if (APPLE)
+        set(CMAKE_MACOSX_RPATH ON)
+        set(CMAKE_INSTALL_RPATH "@loader_path")
+    else()
+        set(CMAKE_INSTALL_RPATH "$ORIGIN/../${CMAKE_INSTALL_LIBDIR}")
```

Comment:
Shouldn't the `$` be escaped? The cmake wiki you linked to mentioned something along those lines.

Reply:
I actually tried to escape it and I obtained unexcepted results. I think this was an issue in old CMake versions, but not anymore. 

Anyways, I also tried to run the executables and now they point correctly to the library path. I think everything is fine right now.